### PR TITLE
Add Timestamps to Events

### PR DESCRIPTION
**Pull request by vdamle**
* Adds a configuration switch at event stream level to enable
`Timestamps` for events
* Uses an extra RPC call to `eth_getBlockByNumber` to get block
timestamp and sets it in the event
* Stores up to 1000 block timestamps in a LRU cache on the event stream
to save an RPC call if multiple events within the stream are emitted
from same block numbers (the lru cache uses inbuilt locking and is
thread safe)

Addresses one of the items in: https://github.com/kaleido-io/ethconnect/issues/68

### CODE REVIEW

File: internal/kldevents/logprocessor.go [hunk 1]
```diff
@@ -112,6 +114,9 @@ func (lp *logProcessor) processLogEntry(subInfo string, entry *logEntry, idx int
 		LogIndex:         strconv.Itoa(idx),
 		batchComplete:    lp.batchComplete,
 	}
+	if lp.stream.spec.Timestamps {
+		result.Timestamp = strconv.Itoa(int(entry.Timestamp))
```
[the remark this answers is not in the document]
Comment (peterbroadhurst) [reply internal/kldevents/logprocessor.go:118]: Saw below it's actually a `uint64` so I'd suggest `strconv.FormatUint(entry.Timestamp, 10)`

File: internal/kldevents/eventstream.go [hunk 2]
```diff
@@ -65,6 +68,7 @@ type StreamInfo struct {
 	RetryTimeoutSec      uint64         `json:"retryTimeoutSec,omitempty"`
 	BlockedRetryDelaySec uint64         `json:"blockedReryDelaySec,omitempty"`
 	Webhook              *webhookAction `json:"webhook,omitempty"`
+	Timestamps           bool           `json:"timestamps,omitempty"` // Include block timestamps in the events generated
```
Comment (peterbroadhurst) [inline internal/kldevents/eventstream.go:71]: I was looking for the change to the `update` function. Not sure if I missed it in the change history?

File: internal/kldevents/logprocessor.go [hunk 3]
```diff
@@ -35,6 +35,7 @@ type logEntry struct {
 	TransactionHash  kldbind.Hash      `json:"transactionHash"`
 	Data             string            `json:"data"`
 	Topics           []*kldbind.Hash   `json:"topics"`
+	Timestamp        uint64            `json:"timestamp,omitifempty"`
```
[the remark this answers is not in the document]
Comment (vdamle) [reply internal/kldevents/logprocessor.go:38]: yeah typo 🤦

File: internal/kldevents/logprocessor.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (vdamle) [reply internal/kldevents/logprocessor.go:118]: Thanks for catching this. I updated the type at some point along the way and missed changing this

File: internal/kldevents/eventstream.go (hunk 2 above)
Comment (vdamle) [reply internal/kldevents/eventstream.go:71]: Duh, missed the `updateStream` change. Will add that

**Pull request merged**